pyaedt/circuit.py

```diff
@@ -1687,3 +1688,404 @@ def import_edb_in_circuit(self, edb_path):
         hfss_3d_layout_model = self.modeler.schematic.add_subcircuit_3dlayout(hfss.design_name)
         hfss.close_project(save_project=False)
         return hfss_3d_layout_model
+
+    @pyaedt_function_handler()
+    def create_tdr_schematic_from_snp(
+        self,
+        touchstone,
+        probe_p_pins,
+        probe_ref_pins=None,
+        termination_pins=None,
+        differential=True,
+        rise_time=30,
+        use_convolution=True,
+        analyze=True,
+        design_name="LNA",
+    ):
+        """Create a new schematic from a touchstone file and automatically setup a TDR transient analysis.
```

Comment:
```suggestion
        """Create a schematic from a Touchstone file and automatically setup a TDR transient analysis.
```
Create means "new" so the word "new" isn't needed.